src/main/flight/position.c

```diff
@@ -183,6 +191,12 @@ void calculateEstimatedAltitude(void)
             zeroedAltitudeCm = baroAltCm; // use Baro if no GPS data, or we want Baro only
         }
     }
+#ifdef USE_RANGEFINDER
+    if (rangefinderAltCm > 0) {
+    // mix it with the range finder 
+        zeroedAltitudeCm = (rangefinderTrust * rangefinderAltCm) + ((1.0 - rangefinderTrust)  * zeroedAltitudeCm);
```

Comment:
This will cause big problems - there will be altitude jump at rangefinder range. rangefinderTrust effectively changes from 0 to user-configured value.

Reply:
If I understand right, the value rangefinder_trust is 0 by default, but once the user sets it into a value, the variable rangefinderTrust  will get this value every time the FC is powered on. If this is right, no jump will happen.

Please explain to me if this is not correct.

Reply:
No rangefinder correction is applied when rangefinder is out of range. This is equivalent to rangefinderTrust = 0. When it gets into range, it is immediately applied with user-specified trust - it will cause jump in estimated altitude (it is unlikely that baro/gps match rangefinder).
This jump will cause problems then altitude is used altitude control loop.

Reply:
Using linear ramp over upper end of range should work good enough. But try it with (poorly tuned) altitude hold